pkg/statistics/handle/ddl/ddl.go

```diff
@@ -206,6 +166,24 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 		// Change id for global stats, since the data has not changed!
 		// Note: This operation will update all tables related to statistics with the new ID.
 		return h.statsWriter.ChangeGlobalStatsID(oldSingleTableID, globalTableInfo.ID)
+	case model.ActionRemovePartitioning:
+		// Change id for global stats, since the data has not changed!
+		// Note: This operation will update all tables related to statistics with the new ID.
+		oldTblID, newSingleTableInfo, droppedPartInfo := s.GetRemovePartitioningInfo()
+		if err := h.statsWriter.ChangeGlobalStatsID(oldTblID, newSingleTableInfo.ID); err != nil {
+			return err
+		}
+
+		// Remove partition stats.
+		for _, def := range droppedPartInfo.Definitions {
+			if err := h.statsWriter.UpdateStatsMetaVersionForGC(def.ID); err != nil {
+				return err
+			}
+		}
+	case model.ActionFlashbackCluster:
+		return h.statsWriter.UpdateStatsVersion()
+	default:
+		return errors.Trace(errors.Errorf("unhandled DDL type %v", s.GetType()))
```

Comment:
I think just an error log would be enough.

Reply:
maybe use `intest` to remind PR author when adding new event types.